custom/icds_reports/reports/awc_reports.py

```diff
@@ -904,6 +904,152 @@ def get_data_for_kpi(filters, date):
     }
 
 
+def get_awc_report_infrastructure(domain, config, month, prev_month, show_test=False):
+    selected_month = datetime(*month)
+    previous_month = datetime(*prev_month)
+
+    def get_data_for_kpi(filters, date):
+        queryset = AggAwcMonthly.objects.filter(
+            month=date, **filters
+        ).values(
+            'aggregation_level'
+        ).annotate(
+            awcs=Sum('num_awcs'),
+            clean_water=Sum('infra_clean_water'),
+            functional_toilet=Sum('infra_functional_toilet'),
+            medicine_kits=Sum('infra_medicine_kits'),
+            infant_weighing_scale=Sum('infra_infant_weighing_scale'),
+            adult_weighing_scale=Sum('infra_adult_weighing_scale')
+        )
+        if not show_test:
+            queryset = apply_exclude(domain, queryset)
+        return queryset
+
+    def get_infa_value(data, prop):
+        value = (data[0][prop] or None) if data else None
+        if value is not None:
+            if value == 1:
+                return _("Available")
+            else:
+                return _("Not Available")
+        else:
+            return _("Data not Entered")
+
+    kpi_data = get_data_for_kpi(config, selected_month.date())
+    kpi_data_prev_month = get_data_for_kpi(config, previous_month.date())
+
+    return {
+        'kpi': [
+            [
+                {
+                    'label': _('Clean Drinking Water'),
+                    'help_text': None,
+                    'percent': percent_diff(
+                        'clean_water',
+                        kpi_data,
+                        kpi_data_prev_month,
+                        'awcs'
+                    ),
+                    'color': 'green' if percent_diff(
+                        'clean_water',
+                        kpi_data,
+                        kpi_data_prev_month,
+                        'awcs'
+                    ) > 0 else 'red',
```

Comment:
all the `percent_diff` calls seem to be called exactly the same twice, it would make this easier to read if they were pulled out into variables.